sale_order_type/models/sale_order.py

```diff
@@ -31,6 +31,12 @@ def onchange_type_id(self):
         self.warehouse_id = self.type_id.warehouse_id
         self.picking_policy = self.type_id.picking_policy
         self.order_policy = self.type_id.order_policy
+        if self.type_id.payment_term_id:
+            self.payment_term = self.type_id.payment_term_id.id
+        if self.type_id.pricelist_id:
+            self.pricelist_id = self.type_id.pricelist_id.id
+        if self.type_id.incoterm_id:
+            self.incoterm = self.type_id.incoterm_id.id
```

Comment:
This method will invoke multiple times the write method of the sale.order object (potentially this can happen up to 6 times). I believe it would be better, as also advised in the official Odoo documentation, to gather all the values that need to be updated in a dictionary and to explicitly invoke the write method once for all in the end. This for performance reasons.